### ISSUE TITLE
Turn caching on TravisCI

### ISSUE THREAD

**Issue by jaredsburrows** [AzureAD/azure-activedirectory-library-for-android#843]
This way you do not download the dependencies each time.

Example: https://github.com/jaredsburrows/android-gif-example/blob/master/.travis.yml#L31

```
before_cache:
  - rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
  - rm -fr $HOME/.gradle/caches/*/plugin-resolution/
cache:
  directories:
    - $HOME/.gradle/caches/
    - $HOME/.gradle/wrapper/
    - $HOME/.android/build-cache
```

**Comment by jaredsburrows** [comment]
Same idea from https://github.com/AzureAD/microsoft-authentication-library-for-android/issues/192

**Comment by weijjia** [comment]
merged with PR #851

**Comment by jaredsburrows** [comment]
Awesome!
